plugins/out_loki/loki.c

```diff
@@ -1081,30 +1077,35 @@ static int get_tenant_id_from_record(struct flb_loki *ctx, msgpack_object *map)
     }
 
     // check if already dynamic_tenant_id is set.
-    if (ctx->dynamic_tenant_id) {
-        cmp_len = flb_sds_len(ctx->dynamic_tenant_id);
+    if (*dynamic_tenant_id != NULL) {
+        cmp_len = flb_sds_len(*dynamic_tenant_id);
+
         if ((rval->o.via.str.size == cmp_len) &&
-            flb_sds_cmp(tmp_str, ctx->dynamic_tenant_id, cmp_len) == 0) {
+            flb_sds_cmp(tmp_str, *dynamic_tenant_id, cmp_len) == 0) {
             // tenant_id is same. nothing to do.
             flb_ra_key_value_destroy(rval);
             flb_sds_destroy(tmp_str);
+
             return 0;
         }
+
         flb_plg_warn(ctx->ins, "Tenant ID is overwritten %s -> %s",
```

Comment:
Does this give us lots of noisy warnings when the incoming records change tenant ID often? 
If the intention is to have a dynamic ID it seems wrong to warn when the value changes.

Reply:
I think the warning is meant to let the user know the current tenant id is changing which "makes sense" to a degree when you have different tenant IDs in a single chunk because the last one is the one under which all the records in that chunk are going to be filed which I think is a design flaw.

I mentioned that in the issue but the thing is this particular warning was already in the code so I don't think we should focus on it unless there's a reason to believe that the change exacerbates this.